www/base/src/app/common/directives/changedetails/changedetails.directive.js

```diff
@@ -8,10 +8,82 @@ class Changedetails {
                 compact: '=?'
             },
             template: require('./changedetails.tpl.jade'),
+            controller: '_changeDetailsController'
         };
     }
 }
 
+class _changeDetails {
```

Comment:
This is a Directive. It is supposed to be reusable, and I think it is reused in the build's change tab.

So here you are adding some location logic to a directive. This is not a good practice.
I think you should rather make a proper page for change detail. You would click on the change, and this would go to that page, and provide the details as the list of builds associated with the change.
This is simular to the builders page, which display the list of builders, and then have their own page for more details.

Please refactor this code with this approach.

Reply:
Should I also have a new page for changeslist instead of the directive?

Reply:
Can I simply have a href from changeslist directive to a new changes page?

Reply:
There is already a page for changelist, which uses this directive. That is fine.
If you change the directive, then the other pages which show a change list will benefit your improvement (i.e. link to the change detail page), which is the point of a directive compared to implementing everything in a single page.

You can use ui-sref directive, which automatically generate the right href for a state change. Please look at builders page as I suggested.

https://github.com/buildbot/buildbot/blob/7658c965b3bedb3e5532446e1352f62fe7c102ab/www/base/src/app/builders/builders.tpl.jade#L42

Reply:
Thanks

Reply:
Thanks